src/dft/descriptor.cxx

```diff
@@ -98,7 +134,50 @@ void descriptor<prec, dom>::get_value(config_param param, ...) {
     va_list vl;
     va_start(vl, param);
     switch (param) {
-        default: break;
+        case config_param::FORWARD_DOMAIN: *va_arg(vl, dft::domain*) = dom; break;
+        case config_param::DIMENSION: *va_arg(vl, std::int64_t*) = values_.rank; break;
+        case config_param::LENGTHS:
+            std::copy(values_.dimensions.begin(), values_.dimensions.end(),
+                      va_arg(vl, std::int64_t*));
+            break;
+        case config_param::PRECISION: *va_arg(vl, dft::precision*) = prec; break;
+        case config_param::FORWARD_SCALE: *va_arg(vl, double*) = values_.fwd_scale; break;
+        case config_param::BACKWARD_SCALE: *va_arg(vl, double*) = values_.bwd_scale; break;
```

Comment:
@fabiomestre Type should match precision of descriptor for forwards and backwards scale.